st2actions/st2actions/cmd/actionrunner.py

```diff
@@ -51,10 +51,18 @@ def _setup():
 
 def _run_worker():
     LOG.info('(PID=%s) Worker started.', os.getpid())
+
+    components = [
+        scheduler.get_scheduler(),
+        worker.get_worker()
+    ]
+
     try:
-        worker.work()
+        [component.start() for component in components]
```

Comment:
It create an array for now reason. But more importantly list comprehension is not the right sematic - the goal is to run a for-loop so causes confusion. Likely that some linter might while as there is no assignment.